### PR TITLE
Geo-compose: Remove ViewpointOperations

### PR DESCRIPTION
**Pull request by hud10837**
Moves the methods of setting the viewpoint on MapView and SceneView from the ViewpointOperation strategy to calling methods directly on the proxy classes. Note that a viewpoint parameter may be added in a future PR.

Also modifies the microapps to match the new strategy. The MapView microapp has been changed to use multiple locations like the SceneView one.

### CODE REVIEW

File: microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt [hunk 1]
```diff
@@ -113,63 +114,124 @@ fun MainScreen() {
...
         }
-    }
+    ),
+
+    SET_VIEWPOINT_CENTER(
+        "setViewpointCenter",
+        { mapViewProxy ->
+            mapViewProxy.setViewpointCenter(
+                // Catalina
+                Point(
```
Comment (puneet-pdx) [inline microapps/MapViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/mapviewsetviewpointapp/screens/MainScreen.kt:186]: this looks better at these coordinates
 Point(
                    -118.419710,
                    33.432245,
                    SpatialReference.wgs84()
                ),
                1000000.0
            )

File: microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt [hunk 2]
```diff
@@ -93,88 +105,117 @@ fun MainScreen() {
...
+        { sceneViewProxy ->
+            sceneViewProxy.setBookmark(
+                Bookmark(
+                    "Rotterdam",
+                    Viewpoint(Point(4.4777, 51.9244, SpatialReference.wgs84()), 170000.0)
+                )
+            )
+        }
+    )
+}
```
Comment (puneet-pdx) [inline microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt:221]: add newline

File: microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt [hunk 3]
```diff
@@ -77,7 +82,14 @@ fun MainScreen() {
 
                     SetViewpointDropdownMenu(
                         expanded = actionsExpanded,
-                        onSetViewpointOperation = { sceneViewpointOperation = it },
+                        onSelectMethod = {
+                            actionsExpanded = false
+                            coroutineScope.launch {
+                                showProgressIndicator = true
```
Comment (puneet-pdx) [inline microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt:88]: I am still seeing this [issue](https://devtopia.esri.com/runtime/kotlin/issues/3488). Is it working for you?

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoViewProxy.kt [hunk 4]
```diff
@@ -239,6 +244,49 @@ public sealed class GeoViewProxy(className: String) {
...
+     * A success result with a value of false may indicate the operation was cancelled.
+     * @since 200.4.0
+     */
+    public suspend fun setBookmark(bookmark: Bookmark): Result<Boolean> {
+        return geoView?.setBookmark(bookmark)
+            ?: Result.failure(IllegalStateException(nullGeoViewErrorMessage))
+    }
+
+    /**
+     * Change the GeoView to the new viewpoint. The viewpoint is updated instantaneously.
```
Comment (puneet-pdx) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoViewProxy.kt:261]:
```suggestion
     * Change the geo view to the new viewpoint. The viewpoint is updated instantaneously.
```

**Review by puneet-pdx — APPROVED** [review]
@hud10837 looks good. Some minor comments

File: microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt (hunk 3 above)
Comment (hud10837) [reply microapps/SceneViewSetViewpointApp/app/src/main/java/com/arcgismaps/toolkit/sceneviewsetviewpointapp/screens/MainScreen.kt]: I do see it now. I've removed the progress indicator from this app

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoViewProxy.kt [hunk 5]
```diff
@@ -239,6 +244,49 @@ public sealed class GeoViewProxy(className: String) {
...
+     */
+    public suspend fun setViewpointAnimated(
+        viewpoint: Viewpoint,
+        duration: Duration = 0.25.seconds
+    ): Result<Boolean> {
+        return geoView?.setViewpointAnimated(
+            viewpoint,
+            duration.toDouble(DurationUnit.SECONDS).toFloat()
+        )
+            ?: Result.failure(IllegalStateException(nullGeoViewErrorMessage))
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/GeoViewProxy.kt:287]: It's better to move the line above up by one

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewProxy.kt [hunk 6]
```diff
@@ -95,4 +100,99 @@ public class MapViewProxy : GeoViewProxy("MapView") {
...
+     * Animate the map view to the center point and scale.
+     *
+     * @param center the location on which the map should be centered
+     * @param scale the new map scale
+     * @return a [Result] indicating whether the viewpoint was successfully set.
+     * A success result with a value of false may indicate the operation was cancelled.
+     * @since 200.4.0
+     */
+    public suspend fun setViewpointCenter(center: Point, scale: Double? = null): Result<Boolean> {
+        val result = if (scale != null) {
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewProxy.kt:136]: It may be slightly better to `return if`  here.

**Review by changanxian — APPROVED** [review]
LGTM, a few minor comments.

File: toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewProxy.kt [hunk 7]
```diff
@@ -95,4 +100,99 @@ public class MapViewProxy : GeoViewProxy("MapView") {
...
+     * in density-independent pixels
+     * @return a [Result] indicating whether the viewpoint was successfully set.
+     * A success result with a value of false may indicate the operation was cancelled.
+     * @since 200.4.0
+     */
+    public suspend fun setViewpointGeometry(
+        boundingGeometry: Geometry,
+        paddingInDips: Double? = null
+    ): Result<Boolean> {
+        val result = if (paddingInDips != null) {
```
Comment (changanxian) [inline toolkit/geo-compose/src/main/java/com/arcgismaps/toolkit/geocompose/MapViewProxy.kt:159]: It may be slightly better to `return if`  here.

**Pull request merged**